src/psyclone/tests/psyad/main_test.py

```diff
@@ -512,3 +566,47 @@ def test_main_otest_verbose(tmpdir, caplog):
                      "github actions.")
     assert "Writing test harness for adjoint kernel to file" in caplog.text
     assert "/harness.f90" in caplog.text
+
+
+def test_main_otest_lfric(tmpdir, capsys, caplog):
+    ''' Test that the -otest option combined with LFRic API
+    generates the expected adjoint test. '''
+    filename_in = str(tmpdir.join("tl_foo_kernel_mod.f90"))
+    filename_out = str(tmpdir.join("atl_foo_kernel_mod.f90"))
+    harness_out = str(tmpdir.join("atlt_foo_alg_mod.x90"))
+    with open(filename_in, "w", encoding='utf-8') as my_file:
+        my_file.write(TEST_LFRIC_KERNEL)
+    main([filename_in, "-a", "field", "-api", "lfric", "-oad", filename_out,
+          "-otest", harness_out])
+    output, error = capsys.readouterr()
+    assert error == ""
+    assert output == ""
+    with open(harness_out, 'r', encoding='utf-8') as my_file:
+        data = my_file.read()
+    assert "module atlt_foo_alg_mod" in data.lower()
+    assert "subroutine atlt_foo_alg" in data.lower()
+
+
+def test_main_otest_lfric_error_name(tmpdir, capsys, caplog):
+    ''' Test that a bad -otest option combined with LFRic API
+    generates the expected error. '''
+    filename_in = str(tmpdir.join("tl_foo_kernel_mod.f90"))
```

Comment:
As commented earlier, with a slight re-ordering of the validation you should be able to run this test without any valid filenames/files.

Reply:
Addressed in b212e079d51685d8d5d5039504774342e92ef6b1.

Reply:
Please could you tidy this test a bit more: you don't need `filename_out` or `filename_in` any more and the test no longer needs the `tmpdir` fixture. The call to `main` could be:

    harness_out = "/some/path/foo_alg_mod.x90"
    ...
    main(["input.f90", "-a", "field", "-api", "lfric", "-oad", "output.f90", "-otest", harness_out])

Reply:
Addressed in 4d24a0ff8bd638284f06ba18e5fe434f5b625de4.